### PR TITLE
#4083 fixed external link warning when user wants to make a donation

### LINKED ISSUE #4083: Do we really want to warn the user who wants to make a donation?
When I click the donation button (on the left sidebar), and probably because I asked to be warned of external links (which is the default), I get displayed with a warning saying that I'm going online.

While this is technically true, it feels pretty weird to me:
- I'm being warned of something while I want to spend money ... being warned is never something really comfortable, especially when it's about my money, not sure it is the best way to encourage users to give us money
- While browsing a ZIM is expected to be done offline without consuming my data, I don't think that most people would be surprised by being forced to go online
- And in any case, the link opens in the regular phone browsers, so it is quite visible to the user that he is leaving the app and going online

Is this behavior really intentional or only accidental?

### PR DESCRIPTION
**Pull request by viditpawar0**
Fixes #4083 

Fixed external link warning when user wants to make a donation

### CODE REVIEW

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.kt [hunk 1]
```diff
@@ -280,7 +280,7 @@ abstract class CoreMainActivity : BaseActivity(), WebViewProvider {
   }
 
   fun openSupportKiwixExternalLink() {
-    externalLinkOpener.openExternalUrl(KIWIX_SUPPORT_URL.toUri().browserIntent())
+    externalLinkOpener.openExternalUrl(KIWIX_SUPPORT_URL.toUri().browserIntent(), false)
```
Comment (MohitMaliFtechiz) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.kt:283]: @viditpawar0 We have the same functionality for custom apps as well. So in `CustomMainActivity` we have a function `openExternalUrl`. With this function, we are opening the `support_url` which is the same as the Kiwix app so we should refactor the code there as well.

**Review by MohitMaliFtechiz — CHANGES_REQUESTED** [review]
@viditpawar0 We have the `DonationPopup` functionality as well which we have recently added, which shows on the reader screen. So when the user clicks on the "Make a donation" button this method calls https://github.com/kiwix/kiwix-android/blob/9fc6e66ecc00ddcc6b93e958934ff3f2ccbbaa8b/core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt#L1676

According to your existing code, it will work fine with the Kiwix app, but for custom apps, we have overrided this method in  `CustomReaderFragment` to make custom behavior according to configuration. So we have to make the changes there as well so that when while user clicks on the donation button in custom apps this popup will not show.


Apart from these changes: The test cases are falling due to this change specially `topLevelDestinationTest` so can you please fix that?

**Comment by viditpawar0** [comment]
Sure, I'll do that

**Comment by viditpawar0** [comment]
Done

**Comment by viditpawar0** [comment]
All the requested changes have been done.

Summary: -
- made the same fix for custom apps in `CustomMainActivity`
- modified `topLevelDestinationTest`
- fixed the same for `CoreReaderFragment` and `CustomReaderFragment`

**Review by MohitMaliFtechiz — CHANGES_REQUESTED** [review]
@viditpawar0 Thanks for making the changes. Please see comments.

**Comment by viditpawar0** [comment]
Okay, made the required changes

**Review by MohitMaliFtechiz — APPROVED** [review]
@viditpawar0 Thanks for making the changes. @kelson42 LGTM.

**Pull request merged**